source/base/quadrature_lib.cc

```diff
@@ -1739,6 +1746,82 @@ QWitherdenVincentSimplex<dim>::QWitherdenVincentSimplex(
 
 
 
+namespace
+{
+  template <int dim>
+  Quadrature<dim>
+  setup_qiterated_1D(const Quadrature<dim> &, const unsigned int)
+  {
+    Assert(false, ExcInternalError());
+    return Quadrature<dim>();
+  }
+
+
+
+  Quadrature<1>
+  setup_qiterated_1D(const Quadrature<1> &base_quad,
+                     const unsigned int   n_copies)
+  {
+    return QIterated<1>(base_quad, n_copies);
+  }
+} // namespace
+
+
+
+template <int dim>
+QIteratedSimplex<dim>::QIteratedSimplex(const Quadrature<dim> &base_quad,
+                                        const unsigned int     n_copies)
+{
+  switch (dim)
+    {
+      case 1:
+        static_cast<Quadrature<dim> &>(*this) =
+          setup_qiterated_1D(base_quad, n_copies);
+        break;
+      case 2:
+      case 3:
+        {
+          const auto n_refinements =
+            static_cast<unsigned int>(std::log2(n_copies));
+          Assert(std::abs(std::pow(2.0, n_refinements) - n_copies) <
+                   n_copies * 1e-10,
+                 ExcMessage("The number of copies must be a power of 2."));
```

Comment:
Does the rounding always work correctly here or should we add `.5`?
```suggestion
          const auto n_refinements =
            static_cast<unsigned int>(std::log2(n_copies) + .5);
          Assert(std::abs(std::pow(2.0, n_refinements) - n_copies) <
                   n_copies * 1e-10,
                 ExcMessage("The number of copies must be a power of 2."));
```

Reply:
Good point - let me throw in a `std::round` to make sure this does what I think it does.